### PR TITLE
providers: support for Proxmox VE (continuation)

### PR DESCRIPTION
**Pull request by arcln**
Hello, this PR is a continuation of #1790. The original PR is doing good, but we cannot push to it and we would like things to go faster so we decided to make another one to finish it.

This PR includes the commit from @b-, and a fix for https://github.com/coreos/ignition/pull/1790#discussion_r1674311256. I also added release note.

I think this is good to go. Feel free to tell me otherwise.

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ nav_order: 9
 ### Features
 
 - Support partitioning disk with mounted partitions
+- Added Proxmox VE provider
```
Comment (jlebon) [inline docs/release-notes.md:14]:
```suggestion
- Support Proxmox VE
```

Comment (jlebon) [reply docs/release-notes.md:14]: Let's also update `docs/supported-platforms.md` (rendered at https://coreos.github.io/ignition/supported-platforms/).

**Review by jlebon — APPROVED** [review]
Thanks! Some minor docs-related comments, but LGTM otherwise.

Were you able to test this?

**Comment by jlebon** [comment]
Also, minor/optional: let's squash it all to one commit (and add yourself as co-author) ?

**Comment by arcln** [comment]
I tested this PR and it seems to work fine in various situations. 

I patched the release notes and added Promox VE to supported platforms. I also squashed the PR to a single commit.

**Comment by abuisine** [comment]
@jlebon would you be so kind as to have a look ?

**Review by jlebon — APPROVED** [review]
LGTM, thanks!

**Comment by b-** [comment]
Awesome! 🚀

**Comment by abuisine** [comment]
Exciting ❤️

**Comment by genebean** [comment]
Can any of you elaborate on how to put this to use? I am new to FCOS & ignition and may simply not know where I should be looking. Thanks!

**Comment by jlebon** [comment]
@genebean If you're interested in using this in FCOS, the larger effort is tracked in https://github.com/coreos/fedora-coreos-tracker/issues/736 and https://github.com/coreos/fedora-coreos-tracker/issues/1652.

**Comment by lperdereau** [comment]
Hi, do we know when a version will include this cool feature?

**Pull request merged**